beetsplug/subsonicupdate.py

```diff
@@ -25,16 +25,68 @@
 """
 from __future__ import division, absolute_import, print_function
 
-from beets.plugins import BeetsPlugin
-from beets import config
-import requests
-import string
 import hashlib
 import random
+import string
+
+import requests
+
+from beets import config
+from beets.plugins import BeetsPlugin
 
 __author__ = 'https://github.com/maffo999'
 
 
+def build_payload():
+    """ To avoid sending plaintext passwords, authentication will be
+    performed via username, a token, and a 6 random
+    letters/numbers sequence.
+    The token is the concatenation of your password and the 6 random
+    letters/numbers (the salt) which is hashed with MD5.
+    """
+
+    user = config['subsonic']['user'].as_str()
+    password = config['subsonic']['pass'].as_str()
+
+    # Pick the random sequence and salt the password
+    r = string.ascii_letters + string.digits
+    salt = "".join([random.choice(r) for n in range(6)])
+    t = password + salt
+    token = hashlib.md5()
+    token.update(t.encode('utf-8'))
+
+    # Put together the payload of the request to the server and the URL
+    return {
+        'u': user,
+        't': token.hexdigest(),
+        's': salt,
+        'v': '1.15.0',  # Subsonic 6.1 and newer.
+        'c': 'beets'
+    }
+
+
+def formal_url():
+    """ Formats URL to send request to Subsonic
+    DEPRECATED schema, host, port, contextpath; use ${url}
+    """
+
+    host = config['subsonic']['host'].as_str()
+    port = config['subsonic']['port'].get(int)
+
+    context_path = config['subsonic']['contextpath'].as_str()
+    if context_path == '/':
+        context_path = ''
+
+    url = config['subsonic']['url'].as_str()
+    if url and url.endsWith('/'):
+        url = url[:-1]
+
+    if not url:
```

Comment:
I recommend moving all the config accesses to `host`, `port`, etc. below this (or an early `return`) so we don't bother looking up configuration information that we don't need.